iceoryx_utils/test/moduletests/test_cxx_forward_list.cpp

```diff
@@ -0,0 +1,1931 @@
+        ++cnt;
+        if (TESTLISTCAPACITY - 1 == cnt)
+        {
+            // for the last element (insert_after) check for different m_value
+            cnt = DEFAULT_VALUE;
+        }
+    }
+}
+
+TEST_F(forward_list_test, InsertAfterFullElementsListRValue)
+{
+    constexpr int DEFAULT_VALUE{13};
+    int cnt = 0;
+
+    auto iter = sut.before_begin();
+
+    // fill full-1
+    for (uint64_t i = 0; i < TESTLISTCAPACITY - 1; ++i)
+    {
+        iter = sut.emplace_after(iter, i);
+        EXPECT_THAT(sut.size(), Eq(i + 1));
+    }
+
+    sut.insert_after(iter, DEFAULT_VALUE);
+
+    ASSERT_THAT(sut.size(), Eq(TESTLISTCAPACITY));
+    ASSERT_THAT(cTor, Eq(0u));
+    ASSERT_THAT(customCTor, Eq(TESTLISTCAPACITY));
+
+    for (auto& fl : sut)
+    {
+        EXPECT_THAT(fl.m_value, Eq(cnt));
+        ++cnt;
+        if (TESTLISTCAPACITY - 1 == cnt)
+        {
+            // for the last element (insert_after) check for different m_value
+            cnt = DEFAULT_VALUE;
+        }
+    }
+}
+
+TEST_F(forward_list_test, IteratorArrowOperator)
+{
+    constexpr int DEFAULT_VALUE{13};
+
+    ASSERT_THAT(TESTLISTCAPACITY, Ge(10u)); // for the rest of magic numbers to fit
+
+    // fill half
+    for (uint64_t i = 0; i < 5; ++i)
+    {
+        sut.emplace_front(i);
+        EXPECT_THAT(sut.size(), Eq(i + 1));
+    }
+
+    auto iter = sut.cbegin();
+    // 2 increments
+    for (uint64_t i = 0; i < 2; ++i)
+    {
+        ++iter;
+    }
+    sut.insert_after(iter, DEFAULT_VALUE);
+
+    ASSERT_THAT(sut.size(), Eq(6u));
+    ASSERT_THAT(cTor, Eq(0u));
+    ASSERT_THAT(customCTor, Eq(6u));
+
+    iter = sut.cbefore_begin();
+    EXPECT_THAT((++iter)->m_value, Eq(4));
+    EXPECT_THAT((++iter)->m_value, Eq(3));
+    EXPECT_THAT((++iter)->m_value, Eq(2));
+    EXPECT_THAT((++iter)->m_value, Eq(DEFAULT_VALUE));
+    EXPECT_THAT((++iter)->m_value, Eq(1));
+    EXPECT_THAT((++iter)->m_value, Eq(0));
+}
+
+
+TEST_F(forward_list_test, IteratorIncrementOperatorBeyondEnd)
+{
+    constexpr int DEFAULT_VALUE{13};
+
+    // fill
+    sut.push_front(DEFAULT_VALUE);
+
+    auto iter = sut.begin();
+    EXPECT_THAT(iter->m_value, Eq(DEFAULT_VALUE));
+    EXPECT_TRUE((++iter) == sut.cend());
+    EXPECT_TRUE((++iter) == sut.cend());
+}
+
+TEST_F(forward_list_test, ConstIteratorIncrementOperatorBeyondEnd)
+{
+    constexpr int DEFAULT_VALUE{13};
+
+    // fill
+    sut.push_front(DEFAULT_VALUE);
+
+    auto iter = sut.cbegin();
+    EXPECT_THAT(iter->m_value, Eq(DEFAULT_VALUE));
+    EXPECT_TRUE((++iter) == sut.cend());
+    EXPECT_TRUE((++iter) == sut.cend());
+}
+
+
+TEST_F(forward_list_test, IteratorComparisonOfDifferentLists)
+{
+    forward_list<CTorTest, TESTLISTCAPACITY> cut1, cut2;
+    cut1.emplace_front(15842);
+    cut1.emplace_front(1584122);
+    cut1.emplace_front(158432);
+    cut1.emplace_front(158432);
+
+    cut2.emplace_front(1313);
+    cut2.emplace_front(13131);
+
+
+    auto iter_sut1 = cut1.begin();
```

Comment:
snake case but no need to change (I would have just called them iter1 and iter2)